scalyr_agent/log_processing.py

```diff
@@ -810,6 +831,28 @@ def readline(self, current_time=None):
 
         return result
 
+    def __read_extended_line(self, current_time):
+        # read a complete line from our line_matcher
+        next_line = self.__line_matcher.readline(self.__buffer, current_time)
+
+        # Check to see if we allow for extended lines, and if so, then read more pages so that
+        # we can parse an entire extended line.
+        if (
+            len(next_line) == 0
+            and self.__max_extended_line_length > self.__max_line_length
+        ):
+            while (
+                self.__available_buffer_bytes() < self.__max_extended_line_length
```

Comment:
Just trying to understand if we need any additional (time / iteration count based) guard in this while loop to prevent infinite loop or being stuck here for a long time.

I did check all the methods which are called here and it appears it shouldn't be needed, but wanted to double check.

Reply:
I couldn't figure out any way to get stuck here myself, I thought there might be something with the interaction between this timer and the one used when reading from the file but since this timer never resets unless a line is emitted and we move on I don't think there is any real risk.